web_src/js/components/DashboardRepoList.vue

```diff
@@ -251,10 +251,40 @@ const sfc = {
 
   mounted() {
     const el = document.getElementById('dashboard-repo-list');
+    const activeClass = ['button', 'primary'];
     this.changeReposFilter(this.reposFilter);
     $(el).find('.dropdown').dropdown();
     nextTick(() => {
       this.$refs.search.focus();
+      this.$refs.search.addEventListener('keydown', (e) => {
+        const firstItems = document.querySelector('#dashboard-repo-list .repo-owner-name-list a');
+        const activeItem = document.querySelector('#dashboard-repo-list .primary');
+        const rightPage = document.querySelector('#dashboard-repo-list .octicon-chevron-right');
+        const leftPage = document.querySelector('#dashboard-repo-list .octicon-chevron-left');
+        if (activeItem) {
+          switch (e.key) {
+            case 'Enter':
+              activeItem.click();
+              break;
+            case 'ArrowUp':
+              activeItem.classList.remove(...activeClass);
+              if (activeItem.closest('li').previousSibling.querySelector) activeItem.closest('li').previousSibling.querySelector('a').classList.add(...activeClass);
```

Comment:
This is too brittle and likely against Vite best practices, find a better way that does not do DOM mutations please, e.g. state variables.

Reply:
I have modified it, please check again

Thanks